src/playlist-loader.js

```diff
@@ -429,13 +429,22 @@ export default class PlaylistLoader extends EventTarget {
 
     const startingState = this.state;
     const mediaChange = !this.media_ || playlist.id !== this.media_.id;
+    const masterPlaylistRef = this.master.playlists[playlist.id];
 
     // switch to fully loaded playlists immediately
-    if (this.master.playlists[playlist.id].endList ||
+    if (masterPlaylistRef && masterPlaylistRef.endList ||
         // handle the case of a playlist object (e.g., if using vhs-json with a resolved
         // media playlist or, for the case of demuxed audio, a resolved audio media group)
         (playlist.endList && playlist.segments.length)) {
       // abort outstanding playlist requests
+
+      const update = updateMaster(this.master, playlist);
```

Comment:
To update uri references for media playlists. We can get into a state where they are behind with audio only playlists with altenative audio tracks.

Reply:
Is there a better place we can put that logic? Maybe on refreshes we make sure to update everything? It would help prevent the `media` function from having more potential side effects.

Reply:
good point, let me see if there is a better spot.

Reply:
Moved the code to updateMaster where we update the media group playlist with changes from the requested playlist (usually just segments)